packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -86,19 +101,28 @@ class DatePicker extends React.Component {
     };
 
     this.handleChange = this.handleChange.bind(this);
+    this.handleChangeRaw = this.handleChangeRaw.bind(this);
     this.handleOnSelect = this.handleOnSelect.bind(this);
     this.handleOnClickOutside = this.handleOnClickOutside.bind(this);
   }
 
-  handleOnSelect() {
-    this.requestFocus();
+  handleOnSelect(selectedDate, event) {
+    this.releaseFocus();
+
+    if (this.props.onSelect) {
+      this.props.onSelect(event, selectedDate ? selectedDate.format() : '');
```

Comment:
`selectedDate ? selectedDate.format() : ''` 

In what instance would onSelect be triggered without a selectedDate? Should this check be done above in the if statement?

Reply:
Ideally onSelect should only be invoked when selecting a date from the picker. However, react-datepicker has an issue (https://github.com/Hacker0x01/react-datepicker/issues/990) where onSelect is invoked both when selecting a date from the picker as well as manually entering a valid date or clearing the date. If the date is manaully cleared, onSelect would get triggered and selectedDate would be null. I originally had a check in handleOnSelect in DateTimePicker.jsx to ignore the event if it was a manual change. But I've moved that check to here so that consumers of DatePicker doesn't need to handle it.

https://github.com/cerner/terra-core/pull/768/commits/7f704a5dd8211fc1627f14d09ab2fa2b9f92f5ec